Splash/Splash/Details/DetailsViewController.m

```diff
@@ -0,0 +1,166 @@
+    [self.profilePic loadInBackground];
+    self.scoreNames = @[@"💪 Goal ", @"🧼 Bubblescore", @"🔥 Streak ", @"⏳ Avg. Shower Time ", @"🕜 Total Shower Time ", @"🚿 Number of Showers "];
+    self.detailsTableView.delegate = self;
+    self.detailsTableView.dataSource = self;
+    [self.detailsTableView registerClass:[UITableViewHeaderFooterView class] forHeaderFooterViewReuseIdentifier:DetailsHeaderViewIdentifier];
+    self.detailsTableView.sectionHeaderTopPadding = 0;
+    self.removeFriendButton.layer.cornerRadius = 16;
+    self.addFriendButton.layer.cornerRadius = 16;
+    self.challengeButton.layer.cornerRadius = 16;
+    self.onLBButton.layer.cornerRadius = 16;
+    self.addFriendButton.hidden = YES;
+    self.removeFriendButton.hidden = YES;
+    self.challengeButton.hidden = YES;
+    self.onLBButton.hidden = YES;
+    self.current = [PFUser currentUser];
+    self.friends = self.current[@"friends"];
+}
+
+- (void) viewDidAppear:(BOOL)animated {
+    [self checkFriends];
+    [self checkIfUserEqualsCurrent];
+}
+
+#pragma mark - Helper Methods for Displaying Buttons Based on Friend Status
+
+- (void) checkIfUserEqualsCurrent {
+    if ([self.current.objectId isEqual:self.user.objectId]) {
+        self.addFriendButton.hidden = YES;
+        self.removeFriendButton.hidden = YES;
+        self.challengeButton.hidden = YES;
+        self.onLBButton.hidden = NO;
+    }
+}
+
+- (void) checkFriends {
+    self.onLBButton.hidden = YES;
+    if ([self hasUser:self.friends u:self.user]) {
+        self.addFriendButton.hidden = YES;
+        self.removeFriendButton.hidden = NO;
+        self.challengeButton.hidden = NO;
+    } else {
+        self.addFriendButton.hidden = NO;
+        self.challengeButton.hidden = YES;
+        self.removeFriendButton.hidden = YES;
+    }
+}
+
+- (BOOL) hasUser:(NSArray*)my_arr u:(PFUser*)u {
+    for (PFUser* p in my_arr) {
+        DLog(p.objectId);
+        DLog(u.objectId);
+        if ([p.objectId isEqualToString:u.objectId]) {
+            return YES;
+        }
+    }
+    return NO;
+}
+
+#pragma mark - Action Methods for Buttons
+
+- (IBAction)clickAdd:(id)sender {
+    if (![self hasUser:self.friends u:self.user]) {
+        [self.current addObject:self.user forKey:@"friends"];
+        [self.current saveInBackground];
+        DLog(@"ADDED FRIEND");
+    }
+    
+}
+
+- (IBAction)clickRemove:(id)sender {
+}
+
+- (IBAction)clickChallenge:(id)sender {
+}
+
+#pragma mark - Table View Delegate Methods
+
+- (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger)section {
+    UITableViewHeaderFooterView *header = [tableView
+                                           dequeueReusableHeaderFooterViewWithIdentifier:DetailsHeaderViewIdentifier];
+    return header;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return 1;
+}
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 6;
```

Comment:
Let's also this hardcoded into constant variable like 'kNumberOfSections' or any meaningful name you can think of.